repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/irivalidator/IRIValidatorTest.groovy

```diff
@@ -0,0 +1,47 @@
+package features.filters.irivalidator
+
+import framework.ReposeValveTest
+import org.rackspace.deproxy.Deproxy
+
+/**
+ * Created by jcombs on 1/12/15.
+ */
+class IRIValidatorTest extends ReposeValveTest {
+
+    def setupSpec() {
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort)
+
+        def params = properties.defaultTemplateParams
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/irivalidator", params)
+        repose.start(waitOnJmxAfterStarting: false)
+        repose.waitForNon500FromUrl(reposeEndpoint)
+    }
+
+    def cleanupSpec() {
+        deproxy.shutdown()
+        repose.stop()
+    }
+
+    def "When using iri-validator filter, Repose guards the request to the origin services"() {
+        given:
+        def path = "/" + requestpath
+        if (query != null) {
```

Comment:
You can get rid of this ugly if by putting the question mark on the query, and changing the last case from null to an empty string.